### PR TITLE
Add .clvm.hex pre-commit check

### PR DESCRIPTION
**Pull request by altendky**
Quick follows:
- [ ] make the compiler always write with a single trailing EOL
- [ ] make the compiler always use `\n` for all EOL
- [x] make the compilation only happen conditionally on import since it is a chialisp developer ease of life behavior
- [ ] maybe locate compilation temporary files outside of the repository

Draft for:
- [x] Looks in the `build/` directory which seems not great
- [x] Review duplication introduced by this PR compared to `tests/clvm/test_clvm_compilation.py`
- [x] [`can't compile ("defconstant" "AGG_SIG_UNSAFE" 49), unknown operator`](https://github.com/Chia-Network/chia-blockchain/blob/ac140bd9d9f49fcb2b25f331709bdb7450c4acd2/chia/wallet/puzzles/condition_codes.clvm.hex)
- [x] [`can't compile ("my-id"), unknown operator`](https://github.com/Chia-Network/chia-blockchain/blob/ac140bd9d9f49fcb2b25f331709bdb7450c4acd2/chia/wallet/puzzles/create-lock-puzzlehash.clvm.hex)
- [x] Verify that all `.clvm.hex` files have corresponding `.clvm` files
- [x] Give the generated hash files the same treatments as the hex files.
- [x] Resolve `TODO:`s
- [x] Make the bash-only pre-commit change cross platform
- [x] Linter says `Invoke-Expression is used. Please remove Invoke-Expression from script and find other options instead.`
- [x] Merge first:
  - [x] ~https://github.com/Chia-Network/chia-blockchain/pull/12076~
  - [x] https://github.com/Chia-Network/chia-blockchain/pull/12074
  - [x] ~https://github.com/Chia-Network/chia-blockchain/pull/12075~
  - [x] https://github.com/Chia-Network/chia-blockchain/pull/12077

### CODE REVIEW

**Review by trepca — DISMISSED** [review]
this is great, thanks for tidying things up, was very cumbersome to manage hex/sha256

**Review by Quexington — DISMISSED** [review]
Good stuff, thanks

**Comment by altendky** [comment]
Still, broken, sorry...  at least for Windows.  At least we're checking that now.  `:]`

**Review by Quexington — DISMISSED** [review]
Reapproved

File: tests/check_clvm.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+
+def generate_hash_bytes(hex_bytes: bytes) -> bytes:
+    cleaned_blob = bytes.fromhex(hex_bytes.decode("utf-8"))
+    serialize_program = SerializedProgram.from_bytes(cleaned_blob)
+    result = serialize_program.get_tree_hash().hex()
+    return (result + "\n").encode("utf-8")
+
+
+@typing.final
```
Comment (trepca) [inline tests/check_clvm.py:38]: isn't this a Py3.8+ feature?

Comment (altendky) [reply tests/check_clvm.py:38]: Yes, that should come from `typing_extensions`...  Thanks.  Usually tests catch that, but this makes me feel like we're missing some sort of check.

**Review by Quexington — DISMISSED** [review]

**Review by Quexington — APPROVED** [review]

**Review by trepca — APPROVED** [review]

**Pull request merged**